src/validation.cpp

```diff
@@ -5485,6 +5490,22 @@ SnapshotCompletionResult ChainstateManager::MaybeCompleteSnapshotValidation(
     return SnapshotCompletionResult::SUCCESS;
 }
 
+Chainstate& ChainstateManager::GetChainstateForNewBlock(const uint256& blockhash)
+{
+    AssertLockHeld(::cs_main);
+    const auto* pblock{m_blockman.LookupBlockIndex(blockhash)};
```

Comment:
This function could have an early return of `m_ibd_chainstate` if there is no `m_snapshot_chainstate` which should be the base case, i.e. when assumeutxo is not used?